src/pyfaf/queries.py

```diff
@@ -802,16 +803,39 @@ def get_report_by_id(db, report_id):
                       .first())
 
 
-def get_report_by_hash(db, report_hash):
+def get_report_by_hash(db, report_hash, report_os = None):
     """
     Return pyfaf.storage.Report object from pyfaf.storage.ReportHash
     or None if not found.
     """
 
-    return (db.session.query(Report)
-                      .join(ReportHash)
-                      .filter(ReportHash.hash == report_hash)
-                      .first())
+    return find_report(db, report_hash, report_os)
+
+def find_report(db, report_hash, report_os=None ):
+    '''
+    Return pyfaf.storage.Report object by hash and ureport OS data
+    report_os is array with [name, version, architecture]
+    '''
```

Comment:
Why does this has to be different from the rest of the code in this file when it mimics the same logic? Why passing report_os as dictionary when you can pass arguments separately and filter by each of the fields of the dict iff they are passed? Does it only make sense to filter with all of them? Look at get_releases function for an example.

Does it really need to be called differently then the rest of the functions when it does pretty much the same as the others?

Comment string doesn't make any sense btw, the function even does something different than you've described. In comments input variables are enclosed in back-ticks and comment block is followed by blank line to improve readability.